sdk/storage/storage-queue/src/QueueServiceClient.ts

```diff
@@ -276,4 +226,58 @@ export class QueueServiceClient extends StorageClient {
       ...options
     });
   }
+
+  async *listSegments(
+    marker: string | undefined = undefined,
+    options: ServiceListQueuesSegmentOptions = {}
+  ): AsyncIterableIterator<Models.ServiceListQueuesSegmentResponse> {
+    let listQueuesResponse;
+    do {
+      listQueuesResponse = await this.listQueuesSegment(marker, options);
+      marker = listQueuesResponse.nextMarker;
+      yield await listQueuesResponse;
+    } while (marker);
+  }
+
+  async *listItems(
+    options: ServiceListQueuesSegmentOptions = {}
+  ): AsyncIterableIterator<Models.QueueItem> {
+    let marker: string | undefined;
+    for await (const segment of this.listSegments(marker, options)) {
+      yield* segment.queueItems;
+    }
+  }
+
+  public listQueues(options: ServiceListQueuesSegmentOptions = {}) {
+    const iter = this.listItems(options);
+    return {
+      async next() {
+        const item = (await iter.next()).value;
+        return item ? { done: false, value: item } : { done: true };
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      /**
+       *@param {string} [marker] A string value that identifies the portion of
+       *                          the list of queues to be returned with the next listing operation. The
+       *                          operation returns the NextMarker value within the response body if the
+       *                          listing operation did not return all queues remaining to be listed
+       *                          with the current page. The NextMarker value can be used as the value for
+       *                          the marker parameter in a subsequent call to request the next page of list
+       *                          items. The marker value is opaque to the client.
+       *@param {number} [pagesize] Number of items to be returned. Default value is `10`.
+       */
+      byPage: ({
+        marker = undefined,
+        pagesize = 10
```

Comment:
>10 [](start = 19, length = 2)

Does this value align with other languages SDKs?

Reply:
Removed the default value. We can keep it if needed.